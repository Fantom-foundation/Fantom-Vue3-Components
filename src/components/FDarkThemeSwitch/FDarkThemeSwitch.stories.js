import FDarkThemeSwitch from './FDarkThemeSwitch.vue';
import FButton from '../FButton/FButton.vue';

export default {
    title: 'FDarkThemeSwitch',
    component: FDarkThemeSwitch,
};

export const Default = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <FDarkThemeSwitch aria-label="label" />
            <FDarkThemeSwitch label="label" />
        </div>
    `,
});

export const Size = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <FDarkThemeSwitch option-size="large" aria-label="label" />
            <FDarkThemeSwitch aria-label="label" />
            <FDarkThemeSwitch option-size="small" aria-label="label" />
            <FDarkThemeSwitch option-size="mini" aria-label="label" />
        </div>
    `,
});

export const Colors = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <FDarkThemeSwitch :style="blueStyle" aria-label="label" />
            <FDarkThemeSwitch :style="purpleStyle" aria-label="label" />
        </div>
    `,
    data() {
        return {
            blueStyle: {
                '--fdarkthemeswitch-color': '#08209e',
                '--fdarkthemeswitch-color-hsl': '230, 90%, 33%',
            },
            purpleStyle: {
                '--fdarkthemeswitch-color': '#800E99',
                '--fdarkthemeswitch-color-hsl': '289, 83%, 33%',
            },
        };
    },
});

export const Model = () => ({
    components: { FDarkThemeSwitch, FButton },
    template: `
        <div>
            <FDarkThemeSwitch v-model="tglValue" name="tgl_m" aria-label="label" />
            <FButton secondary size="small" @click.native="tglValue = !tglValue">Toggle</FButton>
            <br />
            Value: {{ tglValue }}
        </div>
    `,
    data() {
        return {
            tglValue: true,
        };
    },
});
