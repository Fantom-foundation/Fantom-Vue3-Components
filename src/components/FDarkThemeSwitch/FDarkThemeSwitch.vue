<template>
    <span class="fdarkthemeswitch">
        <FToggleButton v-bind="$attrs" />
    </span>
</template>

<script>
import FToggleButton from '../FToggleButton/FToggleButton.vue';

/**
 * Toggle button with sun and moon icons
 */
export default {
    name: 'FDarkThemeSwitch',

    components: { FToggleButton },

    inheritAttrs: false,
};
</script>

<style lang="scss">
@use 'style';
</style>
