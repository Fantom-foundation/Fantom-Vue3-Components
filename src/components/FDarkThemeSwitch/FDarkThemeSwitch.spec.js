import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import FDarkThemeSwitch from '@/components/FDarkThemeSwitch/FDarkThemeSwitch.vue';

let wrapper = null;

function createWrapper({ propsData = {} } = {}) {
    wrapper = mount(FDarkThemeSwitch, {
        propsData,
    });
}

describe('FDarkThemeSwitch', () => {
    it('should render correctly', () => {
        createWrapper({ propsData: { id: 'foo' } });

        expect(wrapper.element).toMatchSnapshot();
    });
});
