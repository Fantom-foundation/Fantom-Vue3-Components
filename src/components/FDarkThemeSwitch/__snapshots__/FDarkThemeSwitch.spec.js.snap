// Vitest Snapshot v1

exports[`FDarkThemeSwitch > should render correctly 1`] = `
<span
  class="fdarkthemeswitch"
>
  <label
    class="ftogglebutton togglebtn togglebtn-v1"
  >
    <input
      aria-invalid="false"
      class="togglebtn_input"
      id="foo"
      type="checkbox"
      value=""
    />
    <span
      class="togglebtn_btn"
    />
    <span
      class="togglebtn_label"
    >
      
      
      
    </span>
  </label>
</span>
`;
